test/units_kf_6x2x0_vehicle_location.cpp

```diff
@@ -0,0 +1,133 @@
+    column_vector<typename index<first_v<Reference...>>::scalar,
+                  sizeof...(Reference)>,
+    index<Reference>...>;
+
+using state = fcarouge::state<vector<m, m / s, m / s2, m, m / s, m / s2>>;
+using output_t = vector<m, m>;
+using estimate_uncertainty =
+    fcarouge::estimate_uncertainty<ᴀʙᵀ<state::type, state::type>>;
+using process_uncertainty =
+    fcarouge::process_uncertainty<ᴀʙᵀ<state::type, state::type>>;
+using output_uncertainty =
+    fcarouge::output_uncertainty<ᴀʙᵀ<output_t, output_t>>;
+using output_model = fcarouge::output_model<ᴀʙᵀ<output_t, state::type>>;
+using state_transition =
+    fcarouge::state_transition<ᴀʙᵀ<state::type, state::type>>;
+
+template <auto... Reference>
+inline auto output{fcarouge::output<vector<Reference...>>};
+
+//! @test Verifies compatibility with the `mp-units` quantities and units
+//! library for C++ in the case of an algebraic 6x2x0 filter estimating the
+//! vehicle location.
+//!
+//! @details See the sample for details.
+[[maybe_unused]] auto test{[] {
+  kalman filter{
+      state{0. * m, 0. * m / s, 0. * m / s2, 0. * m, 0. * m / s, 0. * m / s2},
+      output<m, m>,
+      estimate_uncertainty{500. * identity<estimate_uncertainty::type>},
+      process_uncertainty{[]() {
+        auto value{identity<process_uncertainty::type>};
+        value.operator()<0, 0>(0.25 * m2);
+        value.operator()<0, 1>(0.5 * m2 / s);
+        value.operator()<0, 2>(0.5 * m2 / s2);
+        value.operator()<1, 0>(0.5 * m2 / s);
+        value.operator()<1, 2>(1 * m2 / s3);
+        value.operator()<2, 0>(0.5 * m2 / s2);
+        value.operator()<2, 1>(1 * m2 / s4);
+        value.operator()<3, 3>(0.25 * m2);
+        value.operator()<3, 4>(0.5 * m2 / s);
+        value.operator()<3, 5>(0.5 * m2 / s2);
+        value.operator()<4, 3>(0.5 * m2 / s);
+        value.operator()<4, 5>(1 * m2 / s3);
+        value.operator()<5, 3>(0.5 * m2 / s2);
+        value.operator()<5, 4>(1 * m2 / s4);
+        return 0.2 * 0.2 * value;
+      }()},
+      output_uncertainty{{9. * m2, 0. * m2}, {0. * m2, 9. * m2}},
+      output_model{[]() {
+        auto value{zero<output_model::type>};
+        value.operator()<0, 0>(1. * m2);
+        value.operator()<1, 3>(1. * m2);
+        return value;
+      }()},
+      state_transition{[]() {
+        auto value{identity<state_transition::type>};
+        value.operator()<0, 1>(1. * m2 / s);
+        value.operator()<0, 2>(0.5 * m2 / s2);
+        value.operator()<1, 2>(1. * m2 / s3);
+        value.operator()<3, 4>(1. * m2 / s);
+        value.operator()<3, 5>(0.5 * m2 / s2);
+        value.operator()<4, 5>(1. * m2 / s3);
+        return value;
+      }()}};
+
+  filter.predict();
+  filter.update(-393.66 * m, 300.4 * m);
+  filter.predict();
+  filter.update(-375.93 * m, 301.78 * m);
+  filter.predict();
+
+  // Verify the example estimated state at 0.1% accuracy.
+  assert(abs(1 - filter.x().operator[]<0>() / (-277.8 * m)) < 0.001 &&

```

Comment:
Pursuing a classic operator approach: after trying a proxy reference, continuation passing style, and quantity_like_traits specialization, I always end up trying to return a different type based on runtime input which is not possible. The best alternative I use is NTTP on access method as seen in Eigen for the block interface. It's compact and readable but less common. This will do for now.